index.js

```diff
@@ -20,35 +22,293 @@ class J5sExecutor extends Executor {
+     * @param  {Object} options.ecosystem                              Screwdriver Ecosystem
+     * @param  {Object} options.ecosystem.api                          Routable URI to Screwdriver API
+     * @param  {Object} options.ecosystem.store                        Routable URI to Screwdriver Store
+     * @param  {String} options.jenkins.host                           Jenkins hostname to make requests to
+     * @param  {Number} [options.jenkins.port=8080]                    Jenkins port to make requests to
+     * @param  {String} [options.jenkins.username='screwdriver']       Jenkins username
+     * @param  {String} options.jenkins.password                       Jenkins password/token
+     * @param  {String} [options.jenkins.nodeLabel='screwdriver']      Node labels of Jenkins slaves
+     * @param  {String} [options.docker.command='docker']              THe path to the docker command
+     * @param  {String} [options.docker.launchVersion='stable']        Launcher container version to use
+     * @param  {String} [options.docker.prefix='']                     Prefix to all container names
+     * @param  {String} [options.docker.memory='4g']                   Memory limit (docker run `--memory` option)
+     * @param  {String} [options.docker.memoryLimit='6g']              Memory limit include swap (docker run `--memory-swap` option)
+     * @param  {String} [options.buildScript]                          Shell script to start the job
+     * @param  {String} [options.cleanupScript='']                     Shell script to clean up the job
+     * @param  {Number} [options.cleanupTimeLimit=20]                  Time to destory the job(seconds)
+     * @param  {Number} [options.cleanupWatchInterval=2]               Interval to detect the stopped job (seconds)
+     * @param  {String} [options.fusebox]                              Options for the circuit breaker (https://github.com/screwdriver-cd/circuit-fuses)
      */
     constructor(options) {
         super();
-        this.host = options.host;
-        this.username = options.username;
-        this.password = options.password;
+        this.ecosystem = options.ecosystem;
+        this.host = options.jenkins.host;
+        this.port = options.jenkins.port || '8080';
+        this.username = options.jenkins.username || 'screwdriver';
+        this.password = options.jenkins.password;
+        this.nodeLabel = options.jenkins.nodeLabel || 'screwdriver';
+        this.dockerCommand = (options.docker && options.docker.command) || 'docker';
+        this.launchVersion = (options.docker && options.docker.launchVersion) || 'stable';
+        this.prefix = (options.docker && options.docker.prefix) || '';
+        this.memory = (options.docker && options.docker.memory) || '4g';
+        this.memoryLimit = (options.docker && options.docker.memoryLimit) || '6g';
+
+        this.buildScript = options.buildScript;
+        this.cleanupScript = options.cleanupScript || '';
```

Comment:
If the `config.buildScript` option is provided, the executor run the command instead of docker, so `this.buildScript` is also used as a flag. (I think this may be a bit confusing.)

I changed to set a empty string to `this.buildScript` like `this.cleanupScript`, and fixed the if condition in the `_taskScript` method.
```
if (this.buildScript.length !== 0) {
```